### PR TITLE
retreive list of projects from redcap

### PR DESCRIPTION
**Pull request by SimonLab**
see #119 
Get name and description of the projects for /projects endpoint.

Project on Redcap must follow a convention.
- The name of the instrument (form) must start with "project:"
- a "name" field must be defined with the name value define in the annotation field
- a "description" field must be defined with the description value in the annotation field

### CODE REVIEW

File: web/templates/projects/index.html.eex [hunk 1]
```diff
@@ -28,146 +28,12 @@
...
-          <p class="ph3">
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo. Proin sodales pulvinar tempor.
-          </p>
-          <a href="#" class="dtc tc ttu no-underline blue">More About Project #3</a>
-        </div>
-      </div>
-    </div>
+    <%= for {_instrument, project} <- @projects do %>
+      <%= component "project", project: project %>
+    <% end %>
```
Comment (samhstn) [inline web/templates/projects/index.html.eex]: Great idea 👍

**Review by Danwhy — APPROVED** [review]
LGTM

File: web/controllers/redcap/http_client.ex [hunk 2]
```diff
@@ -53,4 +53,35 @@ defmodule ResearchResource.Redcap.HTTPClient do
...
+    {:ok, data} = Poison.Parser.parse(res.body)
+    data
+    |> Enum.filter(fn(question) -> question["form_name"] =~ ~r/project/i end)
+    |> Enum.reduce(%{}, fn(field, acc) ->
+      cond do
+        !acc[field["form_name"]] && (field["field_label"] == "name" || field["field_label"] == "description")  ->
+          Map.put(acc, field["form_name"], Map.put(%{}, field["field_label"], field["field_annotation"]))
+        field["field_label"] == "name" || field["field_label"] == "description" ->
+          Map.put(acc, field["form_name"], Map.put(acc[field["form_name"]], field["field_label"], field["field_annotation"]))
+        true ->
```
Comment (samhstn) [inline web/controllers/redcap/http_client.ex]: @SimonLab could you have a go at tidying up this section, it seems a bit hard to read.
Perhaps abstract out some of the logic
Also I think the:
```elixir
!Bool && (Bool || Bool)
```
pattern is a little hard to read

**Review by samhstn — CHANGES_REQUESTED** [review]
@SimonLab Let me know about my comment.
Let me know if you would like me to have a go at it

**Comment by SimonLab** [comment]
Thanks for the review, @shouston3 I've tried to simplify how we extract the data from Redcap for the projects. Do you mind to have a look at it again? Thanks

**Comment by samhstn** [comment]
This looks nicer @SimonLab 👍 

Quick question though, why are we ignoring the [`web/controllers/redcap/http_client.ex`](https://github.com/research-resource/research_resource/blob/master/coveralls.json#L12) and `web/controllers/qualtrics/http_client.ex` files.

Refactoring may be easier if we have tests for these functions

**Comment by SimonLab** [comment]
We are using ```web/controllers/redcap/in_memory.ex``` for the tests so we never call the the http_client. But we can maybe extract the "helper" functions in http_client in the helpers file, I will create an new issue for that, thanks

**Pull request merged**